### DISCUSSION TITLE
How is polybar fetching current window title using xwindow module?

### DISCUSSION THREAD

**Discussion by beyond9thousand** [polybar/polybar#2791] [Q&A]
What is the script action/commands being used to obtain the current window title?

**Comment by patrick96** [comment]
Polybar doesn't use commands internally for this. It directly communicates with the X server through XCB. Relevant code here:

https://github.com/polybar/polybar/blob/b5764c8a9395465ae4a83ba647e74f94132401f1/src/modules/xwindow.cpp#L46-L58

**Reply by beyond9thousand** [reply]
Is there a way for me to implement the same outside of polybar? I am trying to get the same functionality on eww, but bspc subscribe is not accurate since it does not look for changes unless the focus has been switched

**Reply by patrick96** [reply]
Listening for focus changes and for title changes in the focused window is a non-trivial development effort. But I think xtitle already does what you want: https://github.com/baskerville/xtitle

**Reply by beyond9thousand** [reply]
Yeah i found this after going round about, not realising i had the app all along and was already using it to get single instances
